Tests/scripts/collect_tests_and_content_packs.py

```diff
@@ -1184,19 +1160,53 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, two_be
+    """
+    computes the lowest from version of the modified files and the highest to version of the modified files
+    If no limitation is found- default will be 0.0.0 - 99.99.99
+    Args:
+        all_modified_files_paths: All modified files
+        id_set: the content of the id.set_json
+
+    Returns:
+        The string representation of the lowest from version and highest to version
+    """
+    max_to_version = LooseVersion('0.0.0')
+    min_from_version = LooseVersion('99.99.99')
+    for _, artifacts in id_set.items():
+        for artifact_dict in artifacts:
+            for _, artifact_details in artifact_dict.items():
+                if artifact_details.get('file_path') in all_modified_files_paths:
+                    from_version = artifact_details.get('fromversion')
+                    to_version = artifact_details.get('toversion')
+                    if from_version:
+                        min_from_version = min(min_from_version, LooseVersion(from_version))
+                    if to_version:
+                        max_to_version = max(max_to_version, LooseVersion(to_version))
+    if max_to_version.vstring == '0.0.0':
+        max_to_version = LooseVersion('99.99.99')
+    if min_from_version.vstring == '99.99.99':
+        min_from_version = LooseVersion('0.0.0')
+    logging.debug(f'modified files are {all_modified_files_paths}')
+    logging.debug(f'lowest from version found is {min_from_version}')
+    logging.debug(f'highest to version found is {max_to_version}')
+    return min_from_version.vstring, max_to_version.vstring
+
+
+def create_filter_envs_file(from_version: str, to_version: str, two_before_ga=None, one_before_ga=None, ga=None):
     """Create a file containing all the envs we need to run for the CI"""
     # always run master and PreGA
+    two_before_ga = two_before_ga or AMI_BUILDS.get('TwoBefore-GA', '0').split('-')[0]
+    one_before_ga = one_before_ga or AMI_BUILDS.get('OneBefore-GA', '0').split('-')[0]
+    ga = ga or AMI_BUILDS.get('GA', '0').split('-')[0]
     envs_to_test = {
         'Demisto PreGA': True,
         'Demisto Marketplace': True,
-        'Demisto two before GA': is_any_test_runnable(test_ids=tests, server_version=two_before_ga, conf=conf,
-                                                      id_set=id_set),
-        'Demisto one before GA': is_any_test_runnable(test_ids=tests, server_version=one_before_ga, conf=conf,
-                                                      id_set=id_set),
-        'Demisto GA': is_any_test_runnable(test_ids=tests, server_version=ga, conf=conf, id_set=id_set),
+        'Demisto two before GA': is_runnable_in_server_version(from_version, two_before_ga, to_version),
+        'Demisto one before GA': is_runnable_in_server_version(from_version, one_before_ga, to_version),
+        'Demisto GA': is_runnable_in_server_version(from_version, ga, to_version),
```

Comment:
doesn't it mean we don't start these instances if no test is runnable?

Reply:
No, sometimes files are modified and there are no test playbook that can verify them.

In that case we just want to install the relevant version and see that nothing is broken even if there are no tests to run in it.

I'm not checking the from version and to version of the tests, but of all the modified files.